File: spring-boot-mybatis-demo/src/main/java/com/github/jccode/springbootrestintegrateddemo/service/UserService.java
package com.github.jccode.springbootrestintegrateddemo.service;

import com.github.jccode.springbootrestintegrateddemo.repo.UserMapper;
import com.github.jccode.springbootrestintegrateddemo.model.User;
import com.github.jccode.springbootsample.core.repo.CrudMapper;
import com.github.jccode.springbootsample.core.service.CrudServiceImpl;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class UserService extends CrudServiceImpl<User> {

    @Autowired
    private UserMapper mapper;

    @Autowired
    public UserService(CrudMapper<User> mapper) {
        super(mapper);
    }

}

File: spring-boot-mybatis-demo/src/main/scala/com/github/jccode/springbootrestintegrateddemo/service/UserService.scala
package com.github.jccode.springbootrestintegrateddemo.service

import com.github.jccode.springbootrestintegrateddemo.model.{User, UserCriteria, UserExt}
import com.github.jccode.springbootrestintegrateddemo.repo.UserMapper
import com.github.jccode.springbootsample.core.service.CrudServiceImpl
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import scala.collection.JavaConverters._

@Service
class UserService(@Autowired override val mapper: UserMapper) extends CrudServiceImpl[User](mapper) {

  def findUserWithOrders(example: UserCriteria): Seq[UserExt] = mapper.findUserWithOrders(example).asScala

}
